enterprise/dbcrypt/dbcrypt.go

```diff
@@ -0,0 +1,374 @@
+package dbcrypt
+
+import (
+	"context"
+	"database/sql"
+	"encoding/base64"
+
+	"github.com/lib/pq"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+)
+
+// testValue is the value that is stored in dbcrypt_keys.test.
+// This is used to determine if the key is valid.
+const testValue = "coder"
+
+var (
+	ErrNotEnabled = xerrors.New("encryption is not enabled")
+	b64encode     = base64.StdEncoding.EncodeToString
+	b64decode     = base64.StdEncoding.DecodeString
```

Comment:
self-review: create a follow-up PR to migrate `oauth_access_token` and `oauth_refresh_token` to `BYTEA` instead so that we can dispense with this base64 en/decoding.

Reply:
https://github.com/coder/coder/issues/9580